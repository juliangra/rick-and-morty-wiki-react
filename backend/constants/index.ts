export const GRAPHQL_ENDPOINT = 'https://rickandmortyapi.com/graphql'
