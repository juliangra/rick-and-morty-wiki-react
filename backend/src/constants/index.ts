export const GRAPHQL_ENDPOINT = 'https://rickandmortyapi.com/graphql'
export const JWT_SECRET = process.env.JWT_SECRET || 'not_very_secret'
export const CURSOR_OFFSET = 10
