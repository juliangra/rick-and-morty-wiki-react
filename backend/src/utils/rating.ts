import { RatingValue } from '../generated/graphql'

export const ratingToNumber = (rating: RatingValue) => {
  switch (rating) {
    case RatingValue.One:
      return 1
    case RatingValue.Two:
      return 2
    case RatingValue.Three:
      return 3
    case RatingValue.Four:
      return 4
    case RatingValue.Five:
      return 5
  }
}

export const numberToRating = (rating: number) => {
  switch (rating) {
    case 1:
      return RatingValue.One
    case 2:
      return RatingValue.Two
    case 3:
      return RatingValue.Three
    case 4:
      return RatingValue.Four
    case 5:
      return RatingValue.Five
    default:
      return RatingValue.One
  }
}
