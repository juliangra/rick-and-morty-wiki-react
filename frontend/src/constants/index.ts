export const REMOTE_GRAPHQL_ENDPOINT = ''
export const LOCAL_GRAPHQL_ENDPOINT = 'http://localhost:8081/graphql'
export const JWT_SECRET = process.env.JWT_SECRET || 'not_very_secret'
