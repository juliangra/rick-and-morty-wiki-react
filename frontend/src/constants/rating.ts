export const MAX_RATING = 5
export const FALLBACK_VALUE = Math.ceil(MAX_RATING / 2)
