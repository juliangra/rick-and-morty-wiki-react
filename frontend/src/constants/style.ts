export const HEADER_HEIGHT_PX = 60
