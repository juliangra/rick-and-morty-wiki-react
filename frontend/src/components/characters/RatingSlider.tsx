import { Box, Slider, SliderProps } from '@mantine/core'
import { MAX_RATING } from 'src/constants/rating'
import useRateCharacter from 'src/hooks/ratings/useRateCharacter'
import CustomError from '../common/CustomError'
import CustomLoading from '../common/CustomLoading'

interface RatingSliderProps extends SliderProps {
  characterId: string
  userId: string
  testId?: string
  refetch: () => void
}

const RatingSlider: React.FC<RatingSliderProps> = ({
  characterId,
  userId,
  testId,
  refetch,
  value
}) => {
  const { ratingOnChange, onChange, onChangeEnd, loading, error } = useRateCharacter({
    characterId,
    userId,
    refetch,
    defaultValue: value
  })

  return (
    <Box style={{ marginTop: 24, marginBottom: 24 }}>
      {error && <CustomError description={error} />}
      {loading && <CustomLoading />}
      <Slider
        value={ratingOnChange}
        step={1}
        min={1}
        max={MAX_RATING}
        onChange={onChange}
        onChangeEnd={onChangeEnd}
        data-testid={testId}
        data-cy={testId}
      />
    </Box>
  )
}

export default RatingSlider
