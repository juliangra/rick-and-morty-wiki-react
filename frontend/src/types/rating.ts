import { RatingSchema } from 'src/schemas/rating'
import { z } from 'zod'

export type Rating = z.infer<typeof RatingSchema>
