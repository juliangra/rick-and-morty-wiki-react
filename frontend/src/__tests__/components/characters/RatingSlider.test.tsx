import { createWithProviders, renderWithProviders } from 'src/utils/tests/render'
import { screen, waitFor } from '@testing-library/react'
import RatingSlider from 'src/components/characters/RatingSlider'
import { FALLBACK_VALUE } from 'src/constants/rating'

describe('RatingSlider', () => {
  it('matches snapshot', () => {
    const component = createWithProviders(
      <RatingSlider characterId={''} userId={''} refetch={() => {}} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with fallback value if no value is provided', async () => {
    await renderWithProviders(<RatingSlider characterId={''} userId={''} refetch={() => {}} />)

    await waitFor(() => {
      // This is a workaround due to the value of the `input` element
      // being a nested child of the element with the role `slider`
      expect(screen.getByRole('slider')).toContainHTML(`aria-valuenow="${FALLBACK_VALUE}"`)
    })
  })
})

export {}
