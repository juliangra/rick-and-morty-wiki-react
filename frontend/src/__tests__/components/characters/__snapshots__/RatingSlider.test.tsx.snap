// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingSlider matches snapshot 1`] = `
<div
  className="mantine-1avyp1d"
  style={
    Object {
      "marginBottom": 24,
      "marginTop": 24,
    }
  }
>
  <div
    className="mantine-Slider-root mantine-1tn28tp"
    onKeyDownCapture={[Function]}
    onMouseDownCapture={[Function]}
    tabIndex={-1}
  >
    <div
      className="mantine-d21okt mantine-Slider-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="mantine-Slider-bar mantine-bnigm6"
      />
      <div
        aria-label=""
        aria-valuemax={5}
        aria-valuemin={1}
        aria-valuenow={3}
        className="mantine-Slider-thumb mantine-6xjpl8"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        style={
          Object {
            "left": "50%",
          }
        }
        tabIndex={0}
      />
      <div />
    </div>
    <input
      type="hidden"
      value={3}
    />
  </div>
</div>
`;
